packages/@react-aria/overlays/src/calculatePosition.ts

```diff
@@ -242,31 +250,41 @@ function computePosition(
   } else {
     position[axis] = Math.floor(childOffset[axis] + childOffset[size] + offset);
   }
-
   return position;
 }
 
 function getMaxHeight(
   position: Position,
   boundaryDimensions: Dimensions,
   containerOffsetWithBoundary: Offset,
-  childOffset: Offset,
+  isContainerPositioned: boolean,
   margins: Position,
-  padding: number
+  padding: number,
+  overlayHeight: number,
+  heightGrowthDirection: HeightGrowthDirection
 ) {
-  return position.top != null
+  // TODO: mention the change to the logic in this code, in paticular with how it affectted the tests in calculatePosition
+  // In particular the "left bottom" test case with no viewport offset was calculating a max height of 400 previously because the
+  // overlay's "height growth" was judged to be the downwards direction even though the placement was "left bottom" aligns the bottom of the overlay with the
+  // bottom of the trigger and thus the overlay grows upwards
+  const containerHeight = (isContainerPositioned ? containerOffsetWithBoundary.height : boundaryDimensions[TOTAL_SIZE.height]);
```

Comment:
I found the logic in here to be pretty confusing since it made some assumptions that a overlay with `position.top` specified meant that the overlay was positioned below the trigger and thus expands downwards, something that wasn't true when I had the flip behavior added for submenus. I changed this here to use a `heightGrowthDirection` so that we know in what direction the overlay can grow. I also modified the latter part of the calculation to calculate the overlay's max height from the bottom of the overlay to the top of the boundary instead of from the trigger's top to the top of the boundary since that felt like it made more sense.

Lemme know if I misunderstood some part of this logic here.